### PR TITLE
Add preferred name page for new TRN lookup flow (#526)

### PR DESCRIPTION
**Pull request by wilpow-github**
### Context

We're removing the dependency on Find from ID. This is the second page in the UI ported from Find.

### Changes proposed in this pull request

Add preferred name page

### Checklist

-   [x] Attach to Trello card
-   [ ] Rebased master
-   [ ] Cleaned commit history
-   [ ] Tested by running locally
-   [ ] Reminder created to manually clean any removed app settings post deployment

### CODE REVIEW

File: dotnet-authserver/src/TeacherIdentity.AuthServer/DataAnnotationsExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+using System.ComponentModel.DataAnnotations;
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/DataAnnotationsExtensions.cs:1]: Mostly taken from https://stackoverflow.com/questions/52321148/conditional-validation-in-mvc-net-core-requiredif/59175740#59175740

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Trn/OfficialName.cshtml [hunk 2]
```diff
@@ -29,8 +30,8 @@
                         Have you ever changed your name?
                     </govuk-radios-fieldset-legend>
 
-                    <govuk-radios-item value="false">No</govuk-radios-item>
-                    <govuk-radios-item value="true">
+                    <govuk-radios-item value="@OfficialName.PreviousName.False">No</govuk-radios-item>
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Trn/OfficialName.cshtml:33]: Needed for keeping radio buttons selected through POST

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Trn/PreferredName.cshtml.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    {
+        _linkGenerator = linkGenerator;
+    }
+
+    public string? FirstName => HttpContext.GetAuthenticationState().FirstName;
+    public string? LastName => HttpContext.GetAuthenticationState().LastName;
+
+    // Properties are set in the order that they are declared. Because the value of HasPreferredName
+    // is used in the conditional RequiredIfTrue attribute, it should be set first.
+    [Display(Name = " ")]
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Trn/PreferredName.cshtml.cs:23]: Wasn't able to set name to required page title as this string is evaluated at compile time. Apparently adding a space works though!

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Trn/PreferredName.cshtml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    <div class="govuk-grid-column-two-thirds-from-desktop">
+        <form action="@LinkGenerator.TrnPreferredName()" method="post" asp-antiforgery="true">
+            <h1 class="govuk-heading-xl">@ViewBag.Title</h1>
+
+            <p>You can tell us what you prefer to be called if it’s different from the name on your official documents.</p>
+            <p>For example, if your name is Joanne and you preferred to be called Jo.</p>
+
+            <govuk-radios asp-for="HasPreferredName">
+                <govuk-radios-fieldset>
+                    <govuk-radios-item value="False">Yes</govuk-radios-item>
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Trn/PreferredName.cshtml:21]: Nit: would be good to get the page header within the `govuk-form-group` of the radios component like in the prototype if possible (https://find-a-lost-trn-prototype.herokuapp.com/get-an-identity/preferred-name)

**Review by gunndabad — APPROVED** [review]

**Pull request merged**